tests/components/nest/test_climate_sdm.py

```diff
@@ -1428,6 +1428,9 @@ async def test_thermostat_hvac_mode_failure(
                 "availableModes": ["HEAT", "COOL", "HEATCOOL", "OFF"],
                 "mode": "COOL",
             },
+            "sdm.devices.traits.ThermostatTemperatureSetpoint": {
```

Comment:
Is this change needed?

Reply:
Yes, otherwise the attempt to set the thermostat setpoint  returns   Before it was calling async_set_hvac_mode and throwing the error from there.

```
    async def async_set_temperature(self, **kwargs: Any) -> None:
        """Set new target temperature."""
        hvac_mode = self.hvac_mode
        if kwargs.get(ATTR_HVAC_MODE) is not None:   # <- Test used to take this branch and raise
            hvac_mode = kwargs[ATTR_HVAC_MODE]
            await self.async_set_hvac_mode(hvac_mode)
        low_temp = kwargs.get(ATTR_TARGET_TEMP_LOW)
        high_temp = kwargs.get(ATTR_TARGET_TEMP_HIGH)
        temp = kwargs.get(ATTR_TEMPERATURE)
        if ThermostatTemperatureSetpointTrait.NAME not in self._device.traits:
            return   

```

That return statement should be an error maybe?